### PR TITLE
if there are multiple drafts for an application throw an error. This should not happen.

### PR DESCRIPTION
**Pull request by azizova-leyla**
### Description
If there are multiple drafts for an application - throw an error. This should not happen. 
Also refactored tests to make them clearer.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #<issue_number>; Fixes #<issue_number>; Fixes #<issue_number>...

### CODE REVIEW

**Review by clouser2 — APPROVED** [review]

**Comment by shanemc-goog** [comment]
The title says application but the description says Program. Can you clarify those?

There should never be multiple drafts it's a violation of the System design, if there is that is a system crashing bug typically, and whatever is causing that state should be fixed.

Also what is the issue that this is trying to address?  I don't see a linked issue.

**Comment by azizova-leyla** [comment]
> What is the issue that this is trying to address? I don't see a linked issue.
> 
> It'd be good to understand what puts the system in this incorrect state and address that up front. One issue could be if the "start" landing page is loaded in multiple tabs. We should likely address that case.

This is a hypothetical, I don't actually know how we would get to this state. Brett and I were doing a different change, and  saw this and it seemed that current handling could lead to errors if we ever have more than one draft for whatever reason.

**Comment by azizova-leyla** [comment]
Suggestion from Bion: if this happens, throw 500. Don't try to gracefully recover.

**Review by clouser2 — APPROVED** [review]

File: server/app/repository/ApplicationRepository.java [hunk 1]
```diff
@@ -76,27 +77,37 @@ private Application submitApplicationInternal(
...
-        application.save();
+      List<Application> latestDraft =
+          oldApplications.stream()
+              .filter(app -> app.getLifecycleStage().equals(LifecycleStage.DRAFT))
+              .collect(Collectors.toList());
+
+      if (latestDraft.size() > 1) {
+        throw new RuntimeException(
+            String.format(
+                "Found more than one DRAFT application " + "for applicant %d, program %d.",
```
Comment (bion) [inline server/app/repository/ApplicationRepository.java:88]: nit:
```suggestion
                "Found more than one DRAFT application for applicant %d, program %d.",
```

File: server/app/repository/ApplicationRepository.java [hunk 2]
```diff
@@ -76,27 +77,37 @@ private Application submitApplicationInternal(
...
-          application.setSubmitTimeToNow();
-          completedApplication = Optional.of(application);
-        } else {
-          application.setLifecycleStage(LifecycleStage.OBSOLETE);
-        }
-        application.save();
+      List<Application> latestDraft =
+          oldApplications.stream()
+              .filter(app -> app.getLifecycleStage().equals(LifecycleStage.DRAFT))
+              .collect(Collectors.toList());
```
Comment (bion) [inline server/app/repository/ApplicationRepository.java:83]: suggestion (blocking): prefer immutable collections from guava as a general rule
```suggestion
      ImmutableList<Application> latestDraft =
          oldApplications.stream()
              .filter(app -> app.getLifecycleStage().equals(LifecycleStage.DRAFT))
              .collect(ImmutableList.toImmutableList());
```

File: server/app/repository/ApplicationRepository.java [hunk 3]
```diff
@@ -76,27 +77,37 @@ private Application submitApplicationInternal(
...
+                "Found more than one DRAFT application " + "for applicant %d, program %d.",
+                applicant.id, program.id));
       }
+
       Application application =
-          completedApplication.orElse(new Application(applicant, program, LifecycleStage.ACTIVE));
-      application.setSubmitTimeToNow();
+          !latestDraft.isEmpty()
+              ? latestDraft.get(0)
+              : new Application(applicant, program, LifecycleStage.ACTIVE);
```
Comment (bion) [inline server/app/repository/ApplicationRepository.java:95]: suggestion (blocking): reads a bit easier without the inversion
```suggestion
          latestDraft.isEmpty()
              ? new Application(applicant, program, LifecycleStage.ACTIVE)
              : latestDraft.get(0);
```

**Review by bion — CHANGES_REQUESTED** [review]

**Review by bion — APPROVED** [review]

**Pull request merged**